src/main/java/io/vertx/core/json/JsonCodec.java

```diff
@@ -0,0 +1,11 @@
+package io.vertx.core.json;
+
+/**
+ * This interface represents the Vert.x json representation of any <TARGET_TYPE> <br/>
+ *
+ * To use it in {@code @ModuleGen} annotation you must provide a {@code public static final [JsonCodecType] INSTANCE} to let codegen retrieve an instance of it
+ *
+ * @param <TARGET_TYPE>
+ * @param <JSON_TYPE>
+ */
+public interface JsonCodec <TARGET_TYPE, JSON_TYPE> extends JsonEncoder<TARGET_TYPE, JSON_TYPE>, JsonDecoder<TARGET_TYPE, JSON_TYPE> { }
```

Comment:
let's only keep `JsonCodec` and have the two methods `encode` and `decode` as part of it, we don't need to have the two `JsonEncoder` and `JsonDecoder` parent interfaces

Reply:
We need it because a DataObject can be only decodable/encodable

Reply:
on `@ModuleGen` we use the `JsonCodec` interface and not `JsonEncoder` or `JsonDecoder` so in practice there is no way to do this

Reply:
User cannot use JsonEncoder or JsonDecoder directly (now), but the DataObject can because you should not declare the DataObject codec in `@ModuleGen`

Reply:
why is the data object able to use it ?

Reply:
Because `@DataObject` annotated type loads automatically the codec following the rules explained in the description of this pr: https://github.com/vert-x3/vertx-codegen/pull/248/files#diff-1be71bcbc5baccc51a5edee456f314e5R148